File: Hull_Speed/Hull_Speed/Sailboat.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Hull_Speed
{
    public class Sailboat
    {

        /// <summary>
        /// Gets or sets the name of the boat.
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// Gets or sets the length of the boat. Unit MUST be feets!.
        /// </summary>
        public double Length { get; set; }

        /// <summary>
        /// Calculate the hull speed for the boat.
        /// </summary>
        /// <returns>Hull speed in knots</returns>
        public double Hullspeed()
        {
            return 1.34 * System.Math.Sqrt(Length);
        }

    }
}

File: 21HullSpeedWeb/HullSpeedWeb/Sailboat.js
function Sailboat() {
    // Private member variables
    var name;
    var length;
 
    // Properties (setters and getters)
    this.setName = function(value) {
        name = value;
    }

    this.getName = function(){
        return name;
    }

    this.setLength = function(value) {
        length = value;
    }

    this.getLength = function() {
        return length;
    }

    // Member functions

    // Calculate the hull speed for the boat.
    this.hullspeed = function () {
        return 1.34 * Math.sqrt(length);
    }

}
